rootppl/macros/macros_adaptive.cuh

```diff
@@ -11,6 +11,8 @@
 #define HOST __host__
 #define DEV __device__
 
+#define LOG(x) log(static_cast<floating_t>(x))
```

Comment:
We do not need it and could remove it. However, this is to prevent the models from looking different on CPU and GPU. Doing `log(x)` works on integers and floats on the CPU, but only on floats in CUDA. This is why the `static_cast` is not done on the CPU (answering your second question below). Either we remove this or I'll add a code comment for it. What would you prefer?

Reply:
Oh, I see, that's reasonable. Merging now!